From 2bc3b6fb4e03941a57e4efb5b92b7034e1dcfad6 Mon Sep 17 00:00:00 2001
From: rk <rokups@zoho.com>
Date: Wed, 1 Jan 2025 13:40:52 +0200
Subject: [PATCH] Custom patches.

---
 CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d4b3c1b..21de841 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,6 +172,18 @@ endif()
 
 add_library(spdlog::spdlog ALIAS spdlog)
 
+if (EMSCRIPTEN)
+	target_compile_options(spdlog PRIVATE -sUSE_PTHREADS=1 -pthread)
+endif ()
+
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    target_compile_options(spdlog PUBLIC $<$<COMPILE_LANGUAGE:CXX>:-frtti>)
+elseif (MSVC)
+    if (NOT " /GR " MATCHES CMAKE_CXX_FLAGS)
+        target_compile_options(spdlog PUBLIC $<$<COMPILE_LANGUAGE:CXX>:/GR>)
+    endif ()
+endif ()
+
 set(SPDLOG_INCLUDES_LEVEL "")
 if(SPDLOG_SYSTEM_INCLUDES)
     set(SPDLOG_INCLUDES_LEVEL "SYSTEM")
-- 
2.45.0

